src/plan.rs

```diff
@@ -175,6 +175,86 @@ impl Recipe {
             Recipe::BluesteinsAlgorithm { len, .. } => *len,
         }
     }
+
+    pub fn cost(&self) -> f32 {
+        match self {
+            Recipe::DFT(len) => 0.9 * (*len as f32).powf(2.2),
+            Recipe::Radix4(len) => {
+                let mut cost = 1.2 * (*len as f32).powf(1.2);
+                if *len > SLOWDOWN_LEN {
+                    cost += 10.0 * (*len - SLOWDOWN_LEN) as f32;
+                }
+                cost
+            }
+            Recipe::Butterfly1 => 0.3,
+            Recipe::Butterfly2 => 0.625,
+            Recipe::Butterfly3 => 0.848,
+            Recipe::Butterfly4 => 1.36,
+            Recipe::Butterfly5 => 2.65,
+            Recipe::Butterfly6 => 2.11,
+            Recipe::Butterfly7 => 6.46,
+            Recipe::Butterfly8 => 5.45,
+            Recipe::Butterfly11 => 12.9,
+            Recipe::Butterfly13 => 17.6,
+            Recipe::Butterfly16 => 25.8,
+            Recipe::Butterfly17 => 31.0,
+            Recipe::Butterfly19 => 39.6,
+            Recipe::Butterfly23 => 98.0,
+            Recipe::Butterfly29 => 198.0,
+            Recipe::Butterfly31 => 233.0,
+            Recipe::Butterfly32 => 86.3,
+            Recipe::MixedRadix {
+                left_fft,
+                right_fft,
+            }
+            | Recipe::MixedRadixSmall {
+                left_fft,
+                right_fft,
+            } => {
+                let len = self.len();
+                let inners_cost = left_fft.cost() * right_fft.len() as f32
+                    + right_fft.cost() * left_fft.len() as f32;
+                let mut twiddle_cost = 15.0 + 1.3 * len as f32;
+                if len > SLOWDOWN_LEN {
+                    twiddle_cost += 10.0 * (len - SLOWDOWN_LEN) as f32;
+                }
+                inners_cost + twiddle_cost
+            }
+            Recipe::GoodThomasAlgorithm {
+                left_fft,
+                right_fft,
+            } => {
+                let len = self.len();
+                let inners_cost = left_fft.cost() * right_fft.len() as f32
+                    + right_fft.cost() * left_fft.len() as f32;
+                // This means this algo will never be used, good since there currently a bug somewhere if one inner is Bluestein
```

Comment:
Do you have more details on this bug?

Reply:
I started looking, but didn't get very far yet. It seems to happen if a GoodThomas has a Bluesteins as inner. Then the Bluesteins is given a too short scratch space.

Reply:
With MixedRadix it's fine

Reply:
Interesting. I'm not surprised, copying scratch around is quite complicated in the mixed radix and good-thomas algorithms, so I may have made a typo. I'll make unit tests that specifically test high-scratch inner FFTs.

Reply:
It doesn't cause much trouble currently, as far as I can tell GoodThomas is only faster for short inners, and once they are long enough to switch from GoodThomasSmall to regular GoodThomas, then MixedRadix is the same or faster.

Reply:
The scratch bug should now be fixed